backend/dal/Services/Concrete/ProjectService.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            if (String.IsNullOrWhiteSpace(projectNumber)) throw new ArgumentException("Argument cannot be null, empty or whitespace.", nameof(projectNumber));
+            this.User.ThrowIfNotAuthorized(Permissions.PropertyView);
+
+            // Check if user has the ability to view sensitive properties.
+            var userAgencies = this.User.GetAgencies();
+            var viewSensitive = this.User.HasPermission(Permissions.SensitiveView);
+            var isAdmin = this.User.HasPermission(Permissions.AdminProperties);
+
+            var project = this.Context.Projects
+                .Include(p => p.Status)
+                .Include(p => p.TierLevel)
+                .Include(p => p.Agency)
+                .Include(p => p.Agency.Parent)
+                .Include(p => p.Tasks)
+                .Include(p => p.Tasks).ThenInclude(t => t.Task)
+                .Include(p => p.Properties)
+                .Include(p => p.Properties).ThenInclude(p => p.Parcel)
+                .Include(p => p.Properties).ThenInclude(p => p.Parcel).ThenInclude(p => p.Evaluations)
+                .Include(p => p.Properties).ThenInclude(p => p.Parcel).ThenInclude(p => p.Fiscals)
+                .Include(p => p.Properties).ThenInclude(b => b.Building)
+                .Include(p => p.Properties).ThenInclude(b => b.Building).ThenInclude(b => b.Evaluations)
+                .Include(p => p.Properties).ThenInclude(b => b.Building).ThenInclude(b => b.Fiscals)
+                .AsNoTracking()
+                .FirstOrDefault(p => p.ProjectNumber == projectNumber &&
+                    (isAdmin || userAgencies.Contains(p.AgencyId))) ?? throw new KeyNotFoundException();
+
+            // Remove any sensitive properties from the results if the user is not allowed to view them.
+            if (!viewSensitive)
+            {
+                project?.Properties.RemoveAll(p => p.Parcel.IsSensitive);
+                project?.Properties.RemoveAll(p => p.Building.IsSensitive);
+            }
+            return project;
+        }
+
+        /// <summary>
+        /// Add the specified project to the datasource.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <returns></returns>
+        public Project Add(Project project)
+        {
+            project.ThrowIfNull(nameof(project));
+            this.User.ThrowIfNotAuthorized(Permissions.PropertyAdd);
+
+            var agency = this.User.GetAgency(this.Context) ??
+                throw new NotAuthorizedException("User must belong to an agency before adding projects.");
+
+            project.AgencyId = agency.Id; // Always assign the current user's agency to the project.
+            project.Agency = agency;
+            project.StatusId = 0; // Always start a project as a Draft.
+            project.Status = this.Context.ProjectStatus.Find(project.StatusId);
+            project.TierLevel = this.Context.TierLevels.Find(project.TierLevelId);
```

Comment:
Do you not have to do something similar for the properties?

Reply:
Adding properties to projects isn't completed (although it might have basic functionality).